src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -95,7 +97,11 @@ public InitCodeView generateTestMethodInitCode(
                 .initFields(fields)
                 .suggestedName(Name.from("request"))
                 .build());
-    return buildInitCodeViewFlattened(context, rootNode);
+    if (isFlattened) {
+      return buildInitCodeViewFlattened(context, rootNode);
+    } else {
+      return buildInitCodeViewRequestObject(context, rootNode);
```

Comment:
Done.
Now we have 6 different entry points to generate init code for different cases. I would simplified this in a separate PR.